test/e2e/README.md

```diff
@@ -0,0 +1,363 @@
+# End-to-end test suite
+
+This is the entry point of the TriggerMesh E2E test suite.
+
+## Contents
+
+1. [Overview](#overview)
+1. [Running tests](#running-tests)
+   * [Execution](#execution)
+   * [Inputs](#inputs)
+1. [Package organization](#package-organization)
+1. [Writing tests](#writing-tests)
+   * [Structure](#structure)
+   * [Best practices](#best-practices)
+
+## Overview
+
+This package contains a single Go test called `TestE2e` which runs all the test specs contained in its sub-packages.
+
+Each test is written using the [Ginkgo][ginkgo-docs] testing framework using Behavior-Driven Development style ("BDD").
+
+The [`framework.Famework`](./framework/) abstraction, heavily inspired by [Kubernetes' E2E tests][k8s-e2e], offers a
+convenient way for each test spec to run against its own short-lived Kubernetes namespace, with common setup and cleanup
+tasks executed automatically before and after each of these specs.
+
+## Running tests
+
+### Execution
+
+| :warning: Do not ever interrupt running tests with `Ctrl-C`. When interrupted, Ginkgo only triggers the global `AfterSuite` logic and skips all the individual `AfterEach` blocks declared in running tests. This can leave dangling resources behind that are tedious to clean up manually and can become costly if not terminated quickly. Ref. [onsi/ginkgo#222][ginkgo-issue222].
+| :--- |
+
+While it is possible to run tests using `go test` like standard unit tests, it is recommended to use the `ginkgo` CLI
+tool for running Ginkgo test suites, which offers better control over Gingko-specific parameters, such as the
+parallelism of test specs, the format and verbosity of the reporter's output, etc.
+
+```sh
+# Using the Ginkgo version pinned inside `go.mod`
+go run github.com/onsi/ginkgo/ginkgo
+
+# Or using an executable installed via `go get`
+ginkgo
+```
+
+### Inputs
+
+All tests require a `kubeconfig` file containing credentials of a user or service account with elevated permissions to
+interact with a Kubernetes cluster, unless tests run in a Kubernetes pod, in which case the pod's service account is
+used as a fallback. The path to this file can be set using the `-e2e.kubeconfig` flag, which defaults to the value of
+the `KUBECONFIG` environment variable.
+
+Some tests require more specific input, such as access tokens to interact with third-party APIs (AWS, GitHub, etc.). The
+input method for those tests varies depending on the client that is used. For example, the Go client for AWS reads
+security credentials from the environment and falls back to the standard location of a local file containing shared
+credentials ([ref.][aws-go-session]), while the GitHub client expects an OAuth2 access token to be exported in the
+environment ([ref.][gh-client]). Please refer to documentation of each test for a description of the required inputs.
```

Comment:
nothing wrong here. GitHub keys are documented but I don't think they are being used yet.
I'm in for keeping this as is, since eventually those keys will be needed and we might forget to add the paragraph back.

Reply:
That's true, I should use an example that is relevant to what exists in this repo.

Because this PR only includes one test, would you mind if I update this paragraph in the next PR? E.g. with an example related to Google Cloud.

Reply:
sure!